packages/plugins/slugManager/manager.ts

```diff
@@ -6,41 +6,41 @@ interface ProfileUpdate {
   [key: string]: string | boolean | Meta;
 }
 
+export type SlugFormatType =
```

Comment:
I have a slight preference for typing the config like: 

```js
{
  slugFormat: string | (lang, profile) => string;
  fields?: string[];
}
```

I like that it removes the nested object and reduces the total number of options (don't have to remember generatorFn vs. slugFormat anymore). But was there a reason we might prefer this way instead?